bin/run_unittests.sh

```diff
@@ -57,22 +77,70 @@ fi
 
 randomize_redis_port
 
-set -e
-
 # start Redis on the test port if it's not already running
 redis-server --port $REDIS_PORT &> /dev/null &
 
-./src/libclipper/libclippertests --redis_port $REDIS_PORT
-redis-cli -p $REDIS_PORT "flushall"
-./src/frontends/frontendtests --redis_port $REDIS_PORT
-redis-cli -p $REDIS_PORT "flushall"
-./src/management/managementtests --redis_port $REDIS_PORT
 
-echo "Running Java container tests..."
-cd $DIR
-cd ../containers/java/clipper-java-container
-mvn test
+function run_java_container_tests {
+  echo "Running Java container tests..."
+  cd $DIR
+  cd ../containers/java/clipper-java-container
+  mvn test
+}
+
+function run_rpc_container_tests {
+  echo "Testing container RPC protocol correctness..."
+  cd $DIR
+  cd ../containers/test/
+  ./test_container_rpc.sh $REDIS_PORT
+}
+
+function run_libclipper_tests {
+  echo -e "\nRunning libclipper tests\n\n"
+  ./src/libclipper/libclippertests --redis_port $REDIS_PORT
+}
+
+function run_management_tests {
+  echo -e "\nRunning management tests\n\n"
+  ./src/management/managementtests --redis_port $REDIS_PORT
+}
+
+function run_frontend_tests {
+  echo -e "\nRunning frontend tests\n\n"
+  ./src/frontends/frontendtests --redis_port $REDIS_PORT
+}
+
+function run_all_tests {
+  run_management_tests
+  redis-cli -p $REDIS_PORT "flushall"
+  run_libclipper_tests
+  redis-cli -p $REDIS_PORT "flushall"
+  run_frontend_tests
+  run_java_container_tests
```

Comment:
Great, thanks @Corey-Zumar. In that case, @lucasmoura, do you mind adding a `redis-cli -p $REDIS_PORT "flushall"` line in between each of the run_<suite>_tests calls?